src/dbnode/storage/index.go

```diff
@@ -1669,9 +1677,16 @@ func (i *nsIndex) queryWithSpan(
 	state.Unlock()
 
 	if err != nil {
+		// an unexpected error occurred processing the query, bail without updating timing metrics.
 		return false, err
 	}
 
+	if monitor.parentCanceled {
+		// caller canceled early, don't update the timing metrics since they will be misleading.
+		return false, fmt.Errorf("index query canceled by caller: %s", timeout.String())
+	}
+
+	// update timing metrics even if the query was canceled due to a timeout
```

Comment:
it's pretty important to update these metrics when a timeout occurs, otherwise we could be missing queries that are taking the most time.